### PR TITLE
Fix for `si seq prune -f`

### PR DESCRIPTION
**Pull request by a-tylenda**
**What?**  
After changes in csi-controller and end extending its lifetime after sending `/kill`, the command `si seq prune --force` stopped working. It supposes to remove all Sequences together with their Instances. 

EDIT:
by the way, 
- `csiLifetimeExtensionDelay` renamed to `instanceLifetimeExtensionDelay`
- Moved `instanceLifetimeExtensionDelay` to `sth-config`
- `removeImediatelly` added to `KILL` command data to bypass `instanceLifetimeExtensionDelay` what will end CSIC and remove instance from Host after `runnerExitDelay`
- Add `--instance-lifetime-extension-delay <ms>` to STH options"
- fix CSIC return `undefined` error when instance has been terminated by `KILL`

**Why?**  

**Usage:**

A fast and simple way to remove all the Sequences at once.

**How it works:**

After executing this command `si seq prune --force` we should get an empty array of Instances and Sequences:
- `si seq ls` → `[]`
- `si inst ls` → `[]`

**To verify**

- `yarn build:packages`
- `npm i -g ./dist/cli`
- start `sth` or use beta (setup config)
- `si seq send <package.tar.gz>`
- `si seq start -`
- `si seq prune -f`

### CODE REVIEW

**Review by alicja-gruzdz — APPROVED** [review]

File: packages/host/src/lib/host.ts [hunk 1]
```diff
@@ -776,7 +779,10 @@ export class Host implements IComponent {
     getInstances(): STHRestAPI.GetInstancesResponse {
         this.logger.info("List Instances");
 
-        return Object.values(this.instancesStore).map(csiController => csiController.getInfo());
+        return Object.values(this.instancesStore).filter((csi) => !csi.isRunning).map(csiController => ({
+            id: csiController.id,
+            sequence: csiController.sequence.id,
+        }));
```
Comment (patuwwy) [inline packages/host/src/lib/host.ts:785]: we need info about instances with other states

**Comment by patuwwy** [comment]
Updated

+ Moved csiLifetimeExtensionDelay to STH Config
+ Some changes to allow to bypass `csiLifetimeExtensionDelay` when killilng instance what will remove instance from Host after 12seconds (runnerExitDelay)

**Comment by patuwwy** [comment]
```
$ si seq send packages/reference-apps/endless-names-output.tar.gz && for i in {1..10}; do si seq start - --args=[10000]; done && si seq prune -f

...

Request ok: http://0.0.0.0:8000/api/v1/sequences status: 200 OK
Request ok: http://0.0.0.0:8000/api/v1/instance/6cac6387-3727-40a1-8d75-eb49ca43160c/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/c175aad4-ce3f-4b47-8781-2197b1c2c6ee/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/1ae5b393-31f9-46b2-96e3-b357e8247600/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/be7d191f-b66e-4007-8746-26faf2b71473/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/7c4034f7-48e5-4cc8-a8a9-a44dc96a9faf/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/a1efc605-d5e5-4586-9bc0-32b796975998/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/a50df2df-9cb6-4207-b190-e4af1ca970ff/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/0257cbb8-70ad-473b-9a8b-3d7f7a121947/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/1faa828d-7bdb-4f5b-9e31-fefb07be36d9/_kill status: 202 Accepted
Request ok: http://0.0.0.0:8000/api/v1/instance/287aff70-9a62-4985-87eb-2f9c2ca0b98b/_kill status: 202 Accepted
KILL requested for Instances of Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 10 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 9 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 9 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 9 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 9 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 9 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 9 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 8 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 8 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 8 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 7 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 7 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 7 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 7 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 7 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 7 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 7 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 6 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 5 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequence 5e52be28-2799-4fdd-b9ef-e30dedca1b3b. Waiting for 4 instances to finish...
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Request ok: http://0.0.0.0:8000/api/v1/sequence/5e52be28-2799-4fdd-b9ef-e30dedca1b3b status: 200 OK
Sequences removed successfully.
```

There is still timeout (5 * instances count) and if instances didn't fit in it sequences won't be removed.

**Review by daro1337 — APPROVED** [review]
lgtm

**Pull request merged**